tests/end_to_end/end_to_end.cpp

```diff
@@ -23,6 +24,12 @@ namespace ebpf {
 #pragma warning(pop)
 }; // namespace ebpf
 
+extern "C"
+{
+    ebpf_result_t
```

Comment:
move this prototypeto libs\execution_context\ebpf_core.h

Reply:
I added this here as this is not expected to be used by any product code, except for this test code.

Reply:
product code should never reference "internal" headers like ebpf_core.h.
Don't duplicate prototypes in different places since a mismatch will not be detected at compilation time.

Reply:
If this functionality (enumerate links) is useful, then please expose as an actual API.

Reply:
I can change "ebpf_core_get_next_handle" back to static, and rely on ebpf_object_tracking_terminate() at the end of the test to check that all objects are freed. 

One caveat though -- ebpf_object_tracking_terminate() is void, so we cannot add a REQUIRE check, also ebpf_object_tracking_terminate() calls assert() if any object is not deleted. So the test will fail / assert only for debug builds.

Is the above approach fine?

Reply:
I see libbpf doesn't have a `bpf_link__next()` API although I think it would be useful to be able to enumerate what hook points (attach types) a program is attached to.

Reply:
Point regarding `bpf_program__next() / bpf_map__next()`, these APIs in libbpf do not make any syscall and only iterate over the maps and programs in that bpf_object. These APIs do not iterate over the global list of programs and maps in the system.

We currently have `ebpf_api_get_next_program()` and `ebpf_api_get_next_map()` implemented in our code to iterate over the global list of programs and maps. But as per the API doc, the APIs for getting all the loaded programs and maps are `ebpf_get_program_info()` and `ebpf_get_map_info()` which return a snapshot of all the programs and maps in one IOCTL call.

Reply:
Removed this prototype as test code is not using this API now.